### PR TITLE
.Net: OpenAI V2 - Concepts Migration - Phase 2.0

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

- Added files back to compilation
- Migrated Planners.OpenAI to V2
- Disabled IntegrationTests with Planner V2 to avoid collision with other packages using the OpenAI V1.
- Breaking changes updated folders
   - Concepts
      - Dependency Injection
      - Functions 
         - [FunctionResult_StronglyTyped.cs](https://github.com/microsoft/semantic-kernel/blob/b4a4caa111397d0172bc1e9023907919311df6c0/dotnet/samples/Concepts/Functions/FunctionResult_StronglyTyped.cs) (bigger)
      - Memory
   - LearnResources
      - AIServices

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.OpenAIV2/Extensions/OpenAIMemoryBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using Microsoft.SemanticKernel.Http;
+using Microsoft.SemanticKernel.Memory;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Provides extension methods for the <see cref="MemoryBuilder"/> class to configure OpenAI and AzureOpenAI connectors.
```
Comment (SergeyMenshykh) [inline dotnet/src/Connectors/Connectors.OpenAIV2/Extensions/OpenAIMemoryBuilderExtensions.cs:11]:
```suggestion
/// Provides extension methods for the <see cref="MemoryBuilder"/> class to configure OpenAI connector.
```

File: dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs [hunk 2]
```diff
@@ -136,7 +136,7 @@ public sealed class FunctionCallingStepwisePlanner
             // Look up function in kernel
             foreach (OpenAIFunctionToolCall functionResponse in functionResponses)
             {
-                if (clonedKernel.Plugins.TryGetFunctionAndArguments(functionResponse, out KernelFunction? pluginFunction, out KernelArguments? arguments))
+                if (clonedKernel.Plugins.TryGetOpenAIFunctionAndArguments(functionResponse, out KernelFunction? pluginFunction, out KernelArguments? arguments))
```
Comment (SergeyMenshykh) [inline dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs:139]: We need to rename the method back to TryGetFunctionAndArguments. There's no reason to change it. It was done for Azure connector as well and that extension class will be dropped in favor of this one.

File: dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs [hunk 3]
```diff
@@ -231,7 +231,7 @@ private bool TryGetFunctionResponse(ChatMessageContent chatMessage, [NotNullWhen
         errorMessage = null;
         try
         {
-            functionResponses = openAiChatMessage.GetOpenAIFunctionToolCalls();
+            functionResponses = openAiChatMessage.GetFunctionToolCalls();
```
Comment (SergeyMenshykh) [inline dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs:234]: Same as comment above - we can keep the old name.

File: dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs]: fixed

File: dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs (hunk 3 above)
Comment (SergeyMenshykh) [reply dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs]: fixed

**Review by westey-m — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**